packages/analytics-browser/src/config.ts

```diff
@@ -54,47 +56,90 @@ export class BrowserConfig extends Config implements IBrowserConfig {
   cookieSameSite: string;
   cookieSecure: boolean;
   cookieStorage: Storage<UserSession>;
-  deviceId?: string;
   disableCookies: boolean;
   domain: string;
   includeGclid: boolean;
   includeFbclid: boolean;
   includeReferrer: boolean;
   includeUtm: boolean;
   partnerId?: string;
-  sessionId?: number;
   sessionTimeout: number;
   trackingOptions: TrackingOptions;
-  userId?: string;
+  sessionManager: ISessionManager;
 
   constructor(apiKey: string, userId?: string, options?: BrowserOptions) {
     const defaultConfig = getDefaultConfig();
     super({
       ...options,
       apiKey,
-      optOut: Boolean(options?.optOut),
       storageProvider: options?.storageProvider ?? defaultConfig.storageProvider,
       transportProvider: options?.transportProvider ?? defaultConfig.transportProvider,
     });
+    this.cookieStorage = options?.cookieStorage ?? defaultConfig.cookieStorage;
+    this.sessionTimeout = options?.sessionTimeout ?? defaultConfig.sessionTimeout;
+    this.sessionManager = new SessionManager(this.cookieStorage, {
+      apiKey,
+      sessionTimeout: this.sessionTimeout,
+    });
 
     this.appVersion = options?.appVersion;
     this.cookieExpiration = options?.cookieExpiration ?? defaultConfig.cookieExpiration;
     this.cookieSameSite = options?.cookieSameSite ?? defaultConfig.cookieSameSite;
     this.cookieSecure = options?.cookieSecure ?? defaultConfig.cookieSecure;
-    this.cookieStorage = options?.cookieStorage ?? defaultConfig.cookieStorage;
     this.deviceId = options?.deviceId;
     this.disableCookies = options?.disableCookies ?? defaultConfig.disableCookies;
     this.domain = options?.domain ?? defaultConfig.domain;
     this.includeGclid = options?.includeGclid ?? defaultConfig.includeGclid;
     this.includeFbclid = options?.includeFbclid ?? defaultConfig.includeFbclid;
     this.includeReferrer = options?.includeReferrer ?? defaultConfig.includeReferrer;
     this.includeUtm = options?.includeUtm ?? defaultConfig.includeUtm;
+    this.lastEventTime = this.lastEventTime ?? options?.lastEventTime;
+    this.optOut = Boolean(options?.optOut);
     this.partnerId = options?.partnerId;
     this.sessionId = options?.sessionId;
-    this.sessionTimeout = options?.sessionTimeout ?? defaultConfig.sessionTimeout;
     this.trackingOptions = options?.trackingOptions ?? defaultConfig.trackingOptions;
     this.userId = userId;
   }
+
+  get deviceId() {
+    return this.sessionManager.getDeviceId();
+  }
```

Comment:
this is part of better session/cookie storage management. adding getters and setters to allow

```ts
this.config.userId = 'asdf'
```

to also update the cookie storage

this was previously

```ts
this.config.userId = 'asdf'
updateCookies(this.config)
```

which can result to easily forgetting to update cookies